kernel/aster-nix/src/syscall/prctl.rs

```diff
@@ -6,15 +6,35 @@
 use super::SyscallReturn;
 use crate::{
     prelude::*,
-    process::posix_thread::{PosixThreadExt, MAX_THREAD_NAME_LEN},
-    util::{read_cstring_from_user, write_bytes_to_user},
+    process::{
+        posix_thread::{PosixThreadExt, MAX_THREAD_NAME_LEN},
+        signal::sig_num::SigNum,
+    },
+    util::{read_cstring_from_user, write_bytes_to_user, write_val_to_user},
 };
+
 pub fn sys_prctl(option: i32, arg2: u64, arg3: u64, arg4: u64, arg5: u64) -> Result<SyscallReturn> {
     let prctl_cmd = PrctlCmd::from_args(option, arg2, arg3, arg4, arg5)?;
     debug!("prctl cmd = {:x?}", prctl_cmd);
     let current_thread = current_thread!();
     let posix_thread = current_thread.as_posix_thread().unwrap();
     match prctl_cmd {
+        PrctlCmd::PR_SET_PDEATHSIG(signum) => {
+            let current = current!();
+            current.set_parent_death_signal(signum);
+        }
+        PrctlCmd::PR_GET_PDEATHSIG(write_to_addr) => {
+            let write_val = {
+                let current = current!();
+
+                match current.parent_death_signal() {
```

Comment:
Does the use of the API here cause a race condition? For example, if another thread of the same process resets `parent_death_signal` value between obtaining the current `parent_death_signal` and `write_val_to_user`. If this situation does not occur or is acceptable, I think `parent_death_signal` might be implemented using `AtomicU8`, with the value 0 corresponding to `None`.